### PR TITLE
more accuracy for runs sync

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

**Comment by norkans7** [comment]
@nicpottier please review

**Comment by nicpottier** [comment]
Should write a test to make sure this is behaving the way you expect.. IE, simulate a sync with no batch pending, one that takes more than one batch and the completion of a multi-batch one. There's some complicated logic here and best to test it out in a unit test to make sure it is all working.

File: ureport/backend/rapidpro.py [hunk 1]
```diff
@@ -461,24 +457,33 @@ def pull_results(self, poll, modified_after, modified_before, progress_callback=
...
+                if (latest_synced_obj_time is None and batches_latest is not None) or json_date_to_datetime(latest_synced_obj_time) <= json_date_to_datetime(batches_latest):
+                    latest_synced_obj_time = batches_latest
+
                 # update the time for this poll from which we fetch next time
                 cache.set(Poll.POLL_RESULTS_LAST_PULL_CACHE_KEY % (org.pk, poll.flow_uuid),
                           latest_synced_obj_time, None)
 
                 # clear the saved cursor
-                cache.delete(Poll.POLL_RESULTS_LAST_PULL_CURSOR)
+                cache.delete(Poll.POLL_RESULTS_LAST_PULL_CURSOR % (org.pk, poll.flow_uuid))
```
Comment (norkans7) [inline ureport/backend/rapidpro.py]: this was a bug for not clearing the right key

**Comment by norkans7** [comment]
@nicpottier Added tests please review once more

**Comment by nicpottier** [comment]
goforit

**Pull request merged**